invenio_rdm_migrator/streams/actions/drafts.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio RDM migration drafts row load module."""
+
+from datetime import datetime
+from functools import partial
+
+from ...actions import LoadAction
+from ...load.ids import generate_pk, generate_recid, generate_uuid
+from ...load.postgresql.transactions.operations import Operation, OperationType
+from ..models.files import FilesBucket
+from ..models.pids import PersistentIdentifier
+from ..models.records import RDMDraftMetadata, RDMVersionState
+from ..records.table_generators.parents import generate_parent_rows  # FIXME: move
+from ..records.table_generators.references import (
+    CommunitiesReferencesMixin,
+    PIDsReferencesMixin,
+)
+
+
+class RDMDraftCreateAction(LoadAction, CommunitiesReferencesMixin, PIDsReferencesMixin):
+    """RDM draft creation."""
+
+    name = "create-draft"
+
+    def __init__(
+        self,
+        tx_id,
+        pid,
+        bucket,
+        draft,
+        parent,
+        parents_state,
+        records_state,
+        communities_state,
+        pids_state,
+    ):
```

Comment:
minor/question: If I understand correctly `LoadAction` implementations like this one, are the ones referenced in the `TransformAction.mapped_cls` implementations. If that is the case, I see an issue when the `TrasnformAction.transform()` instantiates and returns `cls.mapped_cls(...)`, and passes state-related arguments (i.e. `parents_state`, `records_state`, etc.), which it probably doesn't have access to at that point (?).

If it does (by design) I guess we're good, though it would still feel a bit weird that we have to pass/prop-drill these global state objects around all the time. 

Some proposals/thoughts on the state:

1. Can we bundle-up all of the states under a common `state` object and just pass this one around? Then the classes that need one of them would just do `state.records`, `state.pids`, etc.
2. Would it make sense that we make state objects "truly" global? i.e. wherever you are you just do `from invenio_rdm_migrator import state` and use it as above?

Reply:
Yes, I fully agree. This is a problem, I've mentioned it in the Zenodo-RDM PR. I'll make them "global" and encapsulate in an object.

Reply:
# ⚠️  THIS IS A BIG CHANGE!

I've implemented it, it is all self contained in c0875c57f2777b181bf2387b46f10152401cfb40. However, it is a very core change and we do not have tests for every single table generator/load. I have fixed all tests and they work. Nonetheless, this might have undesired effects only shown when performing a full migration.